### ISSUE TITLE
Unpin cftime

### ISSUE THREAD

**Issue by bjlittle** [SciTools/iris#3751]
The `cftime` package has been pinned to `1.1.3` as the lastest offering `1.2.0` is causing epoch unit test failures, see https://github.com/SciTools/iris/pull/3750

We require to investigate and unpin asap. Either:

- Fix `iris`, or
- Raise an issue with the `cftime` developers

**Comment by stephenworsley** [comment]
It looks like `num2date` is giving a different value for certain dates when running this code:
```
import cftime
import cf_units
year = 1
hrs_unit = cf_units.Unit(("hours since epoch", calendar=cf_units.CALENDAR_GREGORIAN)
hrs_unit.date2num(cftime.datetime(year,1,1))
```

From experiments, it seems that there is a discrepancy between versions when `year` is between 1 and 1582 and there is no longer a discrepancy for `year = 1583`. For what its worth, 1582 is when the Gregorian calendar was first adopted. I wonder if there is some conversion going on for pre-Gregorian dates.

**Comment by stephenworsley** [comment]
It looks like cftime has just released version 1.2.1 which ought to include a fix for this.
